components/style/workqueue.rs

```diff
@@ -276,21 +277,42 @@ impl<QueueData: Sync, WorkData: Send> WorkQueue<QueueData, WorkData> {
         }
 
         // Spawn threads.
+        let mut thread_handles = vec![];
         for (i, thread) in threads.into_iter().enumerate() {
-            spawn_named(
-                format!("{} worker {}/{}", thread_name, i + 1, thread_count),
-                move || {
+            let handle = thread::Builder::new()
+                .name(format!("{} worker {}/{}", thread_name, i + 1, thread_count))
+                .spawn(move || {
                     thread_state::initialize(state | thread_state::IN_WORKER);
                     let mut thread = thread;
                     thread.start()
-                })
+                });
+            match handle {
+                Ok(handle) => {
+                    thread_handles.push(handle);
+                }
+                Err(err) => {
+                    warn!("Failed spawning thread: {:?}", err);
+                    break;
+                }
+            }
+        }
+
+        if thread_handles.len() != thread_count {
+            // At least one worker thread failed to be created, just close the
+            // rest of them, and return an error.
+            for (i, handle) in thread_handles.into_iter().enumerate() {
+                let _ = infos[i].chan.send(WorkerMsg::Exit);
+                let _ = handle.join();
```

Comment:
Hm... Not really, or at least not in a way that's easier than this (because otherwise the threads will be joined when the function returns). Of course we could carry it with the queue, or adding a method to the handle wrapper so it detached it, but I don't think it's simpler than this.